### PR TITLE
Issue66

### PR DESCRIPTION
**Pull request by r-owen**
A fix for issue #66 plus bugs and warnings found by the pyflakes linter

### CODE REVIEW

**Comment by timj** [comment]
These look good to me. The cleanups are extensive enough that I'm not prepared to push the button without @mjuric or @RobertLuptonTheGood taking a quick look.

**Comment by mjuric** [comment]
Yes, I worry a bit about things like importing classes from modules into global namespaces (e.g., `from .eups import Eups`) -- I thought that was considered to be evil.

**Comment by timj** [comment]
Do you prefer `import eups.Eups`?

**Comment by mjuric** [comment]
Replying to self -- on the other hand, there are already instances of this idiom throughout the code, so that may be fine.

**Comment by mjuric** [comment]
@timj My personal practice is to try not to copy symbols from other modules, but people's tastes differ here.

E.g., gravitate towards:
```
import eups
e = eups.Eups()
```
rather than
```
from .eups import Eups
e = Eups()
```
It makes it easier to tell where the class is truly defined (when eyeballing the code).

It's a matter of taste; as EUPS is already doing the latter in many places, I think it's OK to keep doing it (consistency trumps personal taste, IMHO).

**Comment by mjuric** [comment]
@r-owen Regarding importing modules -- I was referring to https://github.com/RobertLuptonTheGood/eups/pull/69#commitcomment-15435406. My objection is to lines such as:
```
from distrib import builder
```
To me that looks as if you're importing a global variable `builder`, from the module `distrib`. I think a clearer way to do this is with something like:
```
import .distrib.builder as builder
```
That conveys that `builder` is a module.

**Comment by r-owen** [comment]
It is a very common python idiom to import a few symbols from another package into the local package's namespace. it is heavily used in all code I've seen, including much of the eups and LSST code. Admittedly good practice demands that such code also have an `__all__ = [....]` but I didn't try to add that.

Importing the eups package is unwise (unless it absolutely necessary, as is probably the case for one or two of these) because it can easily lead to circular imports, since eups imports many symbols from many of these modules.

An alternative is to use `from . import foo` but this also imports a symbol into the global namespace; it just happens to be a module instead of, say, a function or class. It seems overkill for just retrieving one or two symbols.

**Comment by mjuric** [comment]
@r-owen -- I'm OK with importing classes from other modules; see https://github.com/RobertLuptonTheGood/eups/pull/69#issuecomment-171444642

**Comment by r-owen** [comment]
@mjuric I have no opinion about `import .distrib.builder as builder` vs. `from .distrib import builder` and will be happy either way. We need one of them and I don't think anything else will do as well.

**Comment by mjuric** [comment]
@r-owen -- I spent (probably too much) time to try to see if there's consensus on this. I couldn't find much, except that the way you're proposing to write it is mentioned as being OK and recommended in the last paragraph of 6.4.1 [here](https://docs.python.org/2/tutorial/modules.html). So though I still think it's evil (it conveys less information to the reader), I guess I'm fine with it.

I guess I'm also OK with the change to `()` discussed here https://github.com/r-owen/eups/commit/5955fedd7cc6ecb8b6f7daea1e87c15b743229b7#commitcomment-15435711 -- I still have misgivings about readability and whether your average reader will understand the intent, but I guess it doesn't hurt.

Splitting up the `import foo, bar, baz` into two lines is annoying,  but not worth the bother to fix & rebase (IMHO, @RobertLuptonTheGood may feel passionately either way).

Bottom line, it's +1 for me to merge this, and if the BDFL agrees, I'd go ahead and do that (@RobertLuptonTheGood, chime in when you get a chance).

**Comment by mjuric** [comment]
Merged it so we can release 2.0.1 (otherwise the 2.x series is unusable right now); if there are more changes to be added, we can do them in a separate PR.

**Comment by r-owen** [comment]
Thanks. I had a rebased branch ready to go (I ditched my fork once I got permission to commit directly to the official repo), but I guess we can remove that now.
